app/models/iteration.rb

```diff
@@ -119,16 +73,17 @@ def self.count_graph_status project_id
     end
   end
 
-  # Set the start time and end time
+  # Set the start time and end time like yyyy/mm/dd
   def set_timestamp start_time, end_time
+    return nil if not (/(?<year>\d{4})\/(?<month>\d{1,2})\/(?<day>\d{1,2})/.match(start_time) and /(?<year>\d{4})\/(?<month>\d{1,2})\/(?<day>\d{1,2})/.match(end_time))
     start = (start_time.split('/')).map{|s| s.to_i}
-    startime = DateTime.new(start[0],start[1],start[2])
     endtime = (end_time.split('/')).map{|s| s.to_i}
+    startime = DateTime.new(start[0],start[1],start[2])
     ending = DateTime.new(endtime[0],endtime[1],endtime[2])
-    self.update_attribute(:end_time,ending)
-    self.update_attribute(:start_time,startime)
+    self.update_attributes(:end_time=> ending, :start_time=> startime)
```

Comment:
You can use the new format for dictionaries
`end_time: ending, start_time: starttime`

It's a minor issue though.